src/session_handle.cpp

```diff
@@ -92,12 +93,12 @@ namespace libtorrent {
 	{
 		std::shared_ptr<session_impl> s = m_impl.lock();
 		if (!s) aux::throw_ex<system_error>(errors::invalid_session_handle);
-		dispatch(s->get_context(), [=]() mutable
+		dispatch(s->get_context(), std::bind([=](auto&&... args) mutable
```

Comment:
Note that trick with std::bind isn't required since c++20. It can be rewritten as the following:
```c++
dispatch(s->get_context(), [=, ...args = std::forward<Args>(a)]() mutable
{
    // do something
});
```